spring-session/src/main/java/org/springframework/session/data/gemfire/AbstractGemFireOperationsSessionRepository.java

```diff
@@ -0,0 +1,284 @@
+/*
```

Comment:
Yes.  There are several reasons:

1. I wanted to have a clean separation between the basic CRUD operations implemented in the concrete class (e.g. GemFireOperationsSessionRepository) and the event handling/translation logic that would be common (literally the same) for all concrete classes.  The GemFire cache event to Spring Session event translation is, or rather should be, consistent across all GemFire SessionRepository implementations.  The concrete implementation should only have to focus on implementing the CRUD operations IMO.

2. Also, the AbstractGemFireSessionRepository handles "registering" the GemFireOperationSessionRepository as a "CacheListener" on the appropriate GemFire component without the user having to know which component (i.e. the GemFire Region storing Session state).

3. Next, I have just moved the GemFireSession class (along with a new GemFireSessionAttributes supporting class) to the AbstractGemFireSessionRepository class to simplify future SessionRepository implementations based on GemFire.  For instance, another possible implementation might be to use SD's Repository abstraction (i.e. GemfireRepository) to back the SessionRepository based on GemFire.  This was what Vinicius did in his implementation of Spring Session GemFire support, but his approach was quite brittle in how he registered the repo.

There are subtle other reasons, but the 3 points above capture my main intent.  I had thought about naming this class `GemFireSessionRepositorySupport`, but I have not renamed it yet.